error_registry/src/lib.rs

```diff
@@ -54,6 +56,8 @@ impl<D: Debug> BaseError<D> {
     #[must_use]
     pub fn new(msg: String, error_type: ErrorType, data: Option<D>, status: Option<u32>) -> Self {
```

Comment:
```suggestion
    pub fn new(msg: String, error_type: ErrorType, data: Option<D>) -> Self {
```
if `status` auto generated, remove it from `new` params.
Also change `status` type from Option<u32> to u32